packages/fuels-contract/src/script.rs

```diff
@@ -402,4 +463,344 @@ mod test {
+                .collect();
+
+            assert_eq!(change_outputs, expected_change_outputs);
+        }
+
+        #[test]
+        fn spendable_coins_added_to_input() {
+            // given
+            let asset_ids = [AssetId::default(), AssetId::from([1; 32])];
+
+            let generate_spendable_coins = || {
+                asset_ids
+                    .into_iter()
+                    .enumerate()
+                    .map(|(index, asset_id)| Coin {
+                        amount: (index * 10).into(),
+                        block_created: 1u64.into(),
+                        asset_id: asset_id.into(),
+                        utxo_id: Default::default(),
+                        maturity: 0u64.into(),
+                        owner: Default::default(),
+                        status: CoinStatus::Unspent,
+                    })
+                    .collect::<Vec<_>>()
+            };
+
+            let call = given_a_contract_call();
+
+            // when
+            let (inputs, _) = Script::get_transaction_inputs_outputs(
+                &[call],
+                &random_bech32_addr(),
+                generate_spendable_coins(),
+            );
+
+            // then
+            let inputs_as_signed_coins: HashSet<Input> = inputs[1..].iter().cloned().collect();
+
+            let expected_inputs = generate_spendable_coins()
+                .into_iter()
+                .map(|coin| {
+                    Input::coin_signed(
+                        fuel_tx::UtxoId::from(coin.utxo_id),
+                        coin.owner.into(),
+                        coin.amount.0,
+                        coin.asset_id.into(),
+                        0,
+                        0,
+                    )
+                })
+                .collect::<HashSet<_>>();
+
+            assert_eq!(expected_inputs, inputs_as_signed_coins);
+        }
+
+        #[test]
+        fn variable_outputs_appended_to_outputs() {
+            // given
+            let variable_outputs = [100, 200].map(|amount| {
+                Output::variable(random_bech32_addr().into(), amount, Default::default())
+            });
+
+            let calls = variable_outputs
+                .iter()
+                .cloned()
+                .map(|variable_output| {
+                    given_a_contract_call().with_variable_outputs(vec![variable_output])
+                })
+                .collect::<Vec<_>>();
+
+            // when
+            let (_, outputs) = Script::get_transaction_inputs_outputs(
+                &calls,
+                &random_bech32_addr(),
+                Default::default(),
+            );
+
+            // then
+            let actual_variable_outputs: HashSet<Output> = outputs[2..].iter().cloned().collect();
+            let expected_outputs: HashSet<Output> = variable_outputs.into();
+
+            assert_eq!(expected_outputs, actual_variable_outputs);
+        }
+    }
+
+    fn given_a_contract_call() -> ContractCall {
```

Comment:
I think this could be a method of the `ContractCall`, and that the name should convey that there is underlying random generation.